src/Promise.ts

```diff
@@ -195,7 +195,7 @@ export class PromiseShim<T> implements Thenable<T> {
 					for (let i = 0; i < count; ++i) {
 						callbacks[i].call(null);
 					}
-					callbacks = null;
+					callbacks = [];
```

Comment:
The only concern about this, is that this will generate a new `[]` every time this code path gets hit, versus `null` being a global reference.  Since this could quite easily be heavy hit, I think we need to be careful about the performance characteristics of something like this.  I would add `| null` to the `callback` type instead of allocating a new empty array every time.